orchestrator/cosmos/broadcast.go

```diff
@@ -171,150 +171,102 @@ func (s *peggyBroadcastClient) SendBatchConfirm(
+				BlockHeight:   ev.ERC20DeployedEvent.Raw.BlockNumber,
+				Orchestrator:  s.AccFromAddress().String(),
+				CosmosDenom:   ev.ERC20DeployedEvent.CosmosDenom,
+				TokenContract: ev.ERC20DeployedEvent.TokenContract.Hex(),
+				Name:          ev.ERC20DeployedEvent.Name,
+				Decimals:      uint64(ev.ERC20DeployedEvent.Decimals),
+				Symbol:        ev.ERC20DeployedEvent.Symbol,
+			})
+			evCounter["erc20_deploy"]++
 
-	s.logger.Info().
-		Str("event_nonce", valsetUpdate.EventNonce.String()).
-		Uint64("valset_nonce", valsetUpdate.NewValsetNonce.Uint64()).
-		Int("validators", len(valsetUpdate.Validators)).
-		Interface("powers", valsetUpdate.Powers).
-		Uint64("reward_amount", valsetUpdate.RewardAmount.Uint64()).
-		Str("reward_token", valsetUpdate.RewardToken.Hex()).
-		Msg("oracle observed a valset update event; sending MsgValsetUpdateClaim")
-
-	members := make([]*types.BridgeValidator, len(valsetUpdate.Validators))
-	for i, val := range valsetUpdate.Validators {
-		members[i] = &types.BridgeValidator{
-			EthereumAddress: val.Hex(),
-			Power:           valsetUpdate.Powers[i].Uint64(),
 		}
 	}
 
-	msg := &types.MsgValsetUpdatedClaim{
-		EventNonce:   valsetUpdate.EventNonce.Uint64(),
-		ValsetNonce:  valsetUpdate.NewValsetNonce.Uint64(),
-		BlockHeight:  valsetUpdate.Raw.BlockNumber,
-		RewardAmount: sdk.NewIntFromBigInt(valsetUpdate.RewardAmount),
-		RewardToken:  valsetUpdate.RewardToken.Hex(),
-		Members:      members,
-		Orchestrator: s.AccFromAddress().String(),
-	}
-
-	txResponse, err := s.broadcastClient.SyncBroadcastMsg(msg)
-	if err != nil {
-		s.logger.Err(err).Msg("broadcasting MsgValsetUpdatedClaim failed")
-		return err
-	}
-
-	s.logger.Info().
-		Str("tx_hash", txResponse.TxHash).
-		Str("event_nonce", valsetUpdate.EventNonce.String()).
-		Msg("oracle sent ValsetUpdate event successfully")
-
-	return nil
-}
-
-func (s *peggyBroadcastClient) sendERC20DeployedClaims(event *wrappers.PeggyERC20DeployedEvent) error {
-
 	s.logger.Info().
-		Str("event_nonce", event.EventNonce.String()).
-		Str("token_contract", event.TokenContract.Hex()).
-		Str("cosmos_denom", event.CosmosDenom).
-		Msg("oracle observed an ERC20 deployed event; sending MsgERC20DeployedClaim")
-
-	msg := &types.MsgERC20DeployedClaim{
-		EventNonce:    event.EventNonce.Uint64(),
-		BlockHeight:   event.Raw.BlockNumber,
-		Orchestrator:  s.AccFromAddress().String(),
-		CosmosDenom:   event.CosmosDenom,
-		TokenContract: event.TokenContract.Hex(),
-		Name:          event.Name,
-		Decimals:      uint64(event.Decimals),
-		Symbol:        event.Symbol,
-	}
-
-	txResponse, err := s.broadcastClient.SyncBroadcastMsg(msg)
+		Int("deposit", evCounter["deposit"]).
+		Int("withdraw", evCounter["withdraw"]).
+		Int("valset_update", evCounter["valset_update"]).
+		Int("erc20_deploy", evCounter["erc20_deploy"]).
+		Int("total_claims", len(events)).
+		Msg("oracle observed events. Sending claims")
+
+	txResponse, err := s.broadcastClient.SyncBroadcastMsg(msgs...)
```

Comment:
just to confirm here, gas=auto, correct?

Reply:
We do `CalculateGas`, not sure if that's gas=auto, I think not? Check here pls: https://github.com/umee-network/peggo/blob/main/cmd/peggo/client/cosmos.go#L246

Reply:
Also, is there a max size for the tx? Can we check if we are near that limit so we can chunk the msg list? Although this is something easily solvable by reducing the amount of blocks to process, so Idk if I would bother.

Reply:
Yeah `CalculateGas` will get the gas for you so that's good.

Reply:
> Also, is there a max size for the tx? Can we check if we are near that limit so we can chunk the msg list? Although this is something easily solvable by reducing the amount of blocks to process, so Idk if I would bother.

Technically no, but there is a block gas limit.

Reply:
Awesome then. I think this is resolved 👌